addons/website_event/controllers/main.py

```diff
@@ -353,6 +354,10 @@ def _create_attendees_from_registration_post(self, event, registration_data):
     @http.route(['''/event/<model("event.event"):event>/registration/confirm'''], type='http', auth="public", methods=['POST'], website=True)
     def registration_confirm(self, event, **post):
         registrations = self._process_attendees_form(event, post)
+        event_ticket_ids = {registration['event_ticket_id'] for registration in registrations}
+        event_tickets = request.env['event.event.ticket'].browse(event_ticket_ids)
+        if any(event_ticket.seats_limited and event_ticket.seats_available < len(registrations) for event_ticket in event_tickets):
+            return request.redirect('/event/%s/register?failed_state=failed-seats' % event.id)
```

Comment:
Would add a docstring to explain than rather than relying on the constraint checks on the model, we want to check if there are available seats first to show a nicely formatted error message to the end user.

(If I understood correctly).